ddtrace/internal/telemetry/writer.py

```diff
@@ -74,46 +92,64 @@ def get_headers(self, request):
     """
-    Submits Instrumentation Telemetry events to the datadog agent.
-    Supports v1 of the instrumentation telemetry api
+    Common features of Telemetry services
     """
 
     # telemetry endpoint uses events platform v2 api
-    ENDPOINT_V1 = "telemetry/proxy/api/v2/apmtelemetry"
+    ENDPOINT_V2 = "telemetry/proxy/api/v2/apmtelemetry"
+    # Counter representing the number of events sent by the writer. Here we are relying on the atomicity
+    # of `itertools.count()` which is a CPython implementation detail. The sequence field in telemetry
+    # payloads is only used in tests and is not required to process Telemetry events.
+    _sequence = itertools.count(1)
 
-    def __init__(self):
-        # type: () -> None
-        super(TelemetryWriter, self).__init__(interval=_get_interval_or_default())
+    def __init__(self, interval):
+        # type: (float) -> None
+        super(TelemetryBase, self).__init__(interval=interval)
 
         # _enabled is None at startup, and is only set to true or false
         # after the config has been processed
         self._enabled = None  # type: Optional[bool]
+
         self._events_queue = []  # type: List[Dict]
-        self._integrations_queue = []  # type: List[Dict]
         self._lock = forksafe.Lock()  # type: forksafe.ResetObject
-        self._forked = False  # type: bool
+        forksafe.register(self._fork_writer)
+
         # Debug flag that enables payload debug mode.
         self._debug = asbool(os.environ.get("DD_TELEMETRY_DEBUG", "false"))
-        forksafe.register(self._fork_writer)
 
-        # Counter representing the number of events sent by the writer. Here we are relying on the atomicity
-        # of `itertools.count()` which is a CPython implementation detail. The sequence field in telemetry
-        # payloads is only used in tests and is not required to process Telemetry events.
-        self._sequence = itertools.count(1)
-        self._client = _TelemetryClient(self.ENDPOINT_V1)
+        self._client = _TelemetryClient(self.ENDPOINT_V2)
 
-    def _flush_integrations_queue(self):
-        # type: () -> List[Dict]
-        """Flushes and returns a list of all queued integrations"""
-        with self._lock:
-            integrations = self._integrations_queue
-            self._integrations_queue = []
-        return integrations
+    def add_event(self, payload, payload_type):
+        # type: (Dict[str, Any], str) -> None
+        """
+        Adds a Telemetry event to the TelemetryWriter event buffer
+
+        :param Dict payload: stores a formatted telemetry event
+        :param str payload_type: The payload_type denotes the type of telmetery request.
+            Payload types accepted by telemetry/proxy v1: app-started, app-closing, app-integrations-change
+        """
+        if self._enabled:
+            event = {
+                "tracer_time": int(time.time()),
+                "runtime_id": get_runtime_id(),
+                "api_version": "v1",
```

Comment:
```suggestion
                "api_version": "v2",
```

Reply:
IIRC, we said in other comment (or other comment in different PR) we should keep v1 at this moment (system tests and other concerns about the migration to v2), right?

Reply:
Yes then we should update the headers

Reply:
Nevermind, I was confused by the events platform v2 endpoint